crates/compiler/builtins/roc/Json.roc

```diff
@@ -554,91 +506,589 @@ decodeRecord = \initialState, stepField, finalizer -> Decode.custom \bytes, @Jso
             Ok val -> { result: Ok val, rest: afterRecordBytes }
             Err e -> { result: Err e, rest: afterRecordBytes }
 
-decodeTuple = \initialState, stepElem, finalizer -> Decode.custom \initialBytes, @Json {} ->
-        # NB: the stepper function must be passed explicitly until #2894 is resolved.
-        decodeElems = \stepper, state, index, bytes ->
-            { val: newState, rest: beforeCommaOrBreak } <- tryDecode
-                    (
-                        when stepper state index is
-                            TooLong ->
-                                { rest: beforeCommaOrBreak } <- bytes |> anything |> tryDecode
-                                { result: Ok state, rest: beforeCommaOrBreak }
+# JSON NUMBER PRIMITIVE --------------------------------------------------------
+
+# Takes the bytes for a valid Json number primitive into a RocStr
+#
+# Note that this does not handle leading whitespace, any whitespace must be
+# handled in json list or record decoding.
+#
+# |> List.dropIf \b -> b == '+'
+# TODO ^^ not needed if roc supports "1e+2", this supports
+# "+" which is permitted in Json numbers
+#
+# |> List.map \b -> if b == 'E' then 'e' else b
+# TODO ^^ not needed if roc supports "1E2", this supports
+# "E" which is permitted in Json numbers
```

Comment:
We should probably open a Zulip discussion about what `Str` should support regarding `e`/`E`/`+`/etc. I don't think it's ever been worked through in-depth!

Reply:
I'll add an issue to track, and add to my todo list. I think some level of research would be a good idea to present some options. Actually... I have an idea! 🤔

Reply:
[5213](https://github.com/roc-lang/roc/issues/5213) to track this.